src/main/java/hudson/plugins/ec2/win/winrm/WinRMClient.java

```diff
@@ -176,15 +183,40 @@ private void setupHTTPClient() {
         credsProvider.setCredentials(new AuthScope(AuthScope.ANY_HOST, AuthScope.ANY_PORT), new UsernamePasswordCredentials(username, password));
     }
 
-    private DefaultHttpClient buildHTTPClient() {
-        DefaultHttpClient httpclient = new DefaultHttpClient();
-        if(! (username.contains("\\")|| username.contains("/"))){
+    private HttpClient buildHTTPClient() {
+        HttpClientBuilder builder = HttpClientBuilder.create().setDefaultCredentialsProvider(credsProvider);
+        if(! (username.contains("\\")|| username.contains("/"))) {
             //user is not a domain user
-            httpclient.getAuthSchemes().register(AuthPolicy.SPNEGO,new NegotiateNTLMSchemaFactory());
+            Lookup<AuthSchemeProvider> authSchemeRegistry = RegistryBuilder.<AuthSchemeProvider>create()
+                                                            .register(AuthSchemes.BASIC, new BasicSchemeFactory())
+                                                            .register(AuthSchemes.SPNEGO,new NegotiateNTLMSchemaFactory())
+                                                            .build();
+            builder.setDefaultAuthSchemeRegistry(authSchemeRegistry);
         }
-        httpclient.setCredentialsProvider(credsProvider);
-        httpclient.getParams().setParameter(CoreConnectionPNames.CONNECTION_TIMEOUT, 5000);
-        // httpclient.setHttpRequestRetryHandler(new WinRMRetryHandler());
+        if (useHTTPS) {
+            try {
+                SSLConnectionSocketFactory sslConnectionFactory = new SSLConnectionSocketFactory(
+                                                                  new SSLContextBuilder().loadTrustMaterial(null, new TrustSelfSignedStrategy()).build(),
```

Comment:
Now I'm assuming this is a pure refactoring here, but this is not the greatest idea. There should probably be an admin-controlled config setting to disable hostname verification and whatnot.

Reply:
This is pure refactoring, we should enhance this from a security perspective outside of this PR.